### PR TITLE
Support text-2.0

### PR DESCRIPTION
**Pull request by Bodigrim**
Tested with 

```cabal
packages: .
packages: https://hackage.haskell.org/package/text-2.0/candidate/text-2.0.tar.gz
```

### CODE REVIEW

File: unicode-transforms.cabal [hunk 1]
```diff
@@ -89,7 +89,7 @@ library
 
     -- We depend on a lot of internal modules in text. We keep the upper bound
     -- inclusive of the latest stable version.
-    , text >= 1.1.1 && <= 1.2.4.1
+    , text >=1.1.1 && <=1.2.5.0 || >=2.0 && <2.1
```
Comment (harendra-kumar) [inline unicode-transforms.cabal:92]: Are there more releases likely in 1.2 series?

**Review by harendra-kumar — APPROVED** [review]

File: unicode-transforms.cabal (hunk 1 above)
Comment (Bodigrim) [reply unicode-transforms.cabal:92]: I don't think so, but it is not impossible. E. g., if we discover that some patches are needed for GHC 9.0.2 or 9.2.2.

**Comment by Bodigrim** [comment]
CI failure on GHC 7.10.3 is unrelated to the PR, see https://github.com/haskell/actions/blob/a64ac8cca0d4aec4e3fd4a5fbc9ae3c4953e20aa/setup/src/setup-haskell.ts#L63-L69
I'd suggest to drop GHC 7.10 support, even `bytestring` and `text` have already done so.

**Comment by harendra-kumar** [comment]
>  I'd suggest to drop GHC 7.10 support, even bytestring and text have already done so.

If bytestring and text have dropped then we can drop it too, there is no point in supporting it.

**Pull request merged**